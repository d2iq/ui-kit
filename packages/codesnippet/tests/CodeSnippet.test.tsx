import React from "react";
import { createSerializer } from "@emotion/jest";
import { mount } from "enzyme";
import toJson from "enzyme-to-json";

import { CodeSnippet } from "../";

expect.addSnapshotSerializer(createSerializer());

const snippetContent = "snippet content";

jest.mock("copy-to-clipboard", () => jest.fn());

describe("CodeSnippet", () => {
  it("renders default", () => {
    const component = mount(<CodeSnippet>{snippetContent}</CodeSnippet>);

    expect(toJson(component)).toMatchSnapshot();
  });
  it("renders with copy button", () => {
    const component = mount(
      <CodeSnippet textToCopy={snippetContent}>{snippetContent}</CodeSnippet>
    );

    expect(toJson(component)).toMatchSnapshot();
  });
  it("calls onCopy when copy button is clicked ", () => {
    const onCopyFn = jest.fn();
    const component = mount(
      <CodeSnippet textToCopy={snippetContent} onCopy={onCopyFn}>
        {snippetContent}
      </CodeSnippet>
    );

    expect(onCopyFn).not.toHaveBeenCalled();
    component.find('[role="button"] svg').simulate("click");
    expect(onCopyFn).toHaveBeenCalled();
  });
});
