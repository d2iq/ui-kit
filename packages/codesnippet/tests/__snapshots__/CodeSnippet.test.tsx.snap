// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeSnippet renders default 1`] = `
.emotion-6 {
  overflow: auto;
  white-space: pre;
}

.emotion-5 {
  overflow: auto;
  white-space: pre;
  padding: 16px;
}

.emotion-4 {
  background-color: #1B2029;
  background-repeat: no-repeat;
  overflow: auto;
  white-space: pre;
  padding: 16px;
}

.emotion-3 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  height: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
}

.emotion-3 > div {
  width: auto;
}

.emotion-3 > *:not(:first-child) {
  padding-left: 0;
  padding-top: 0;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
}

.emotion-1 {
  font-family: 'Menlo','Bitstream Vera Sans Mono','DejaVu Sans Mono','Monaco','Consolas',monospace;
}

.emotion-0 {
  font-family: 'Menlo','Bitstream Vera Sans Mono','DejaVu Sans Mono','Monaco','Consolas',monospace;
  line-height: 1.4;
  font-weight: 500;
  font-size: 14px;
  color: var(--themeTextColorPrimaryInverted,#FFFFFF);
  fill: var(--themeTextColorPrimaryInverted,#FFFFFF);
  text-align: inherit;
}

<CodeSnippet>
  <SpacingBox
    bgColor="#1B2029"
    className="emotion-6"
    side="all"
    spacingSize="m"
    tag="div"
  >
    <Box
      bgColor="#1B2029"
      bgImageOptions={
        Object {
          "position": undefined,
          "repeat": undefined,
          "size": undefined,
        }
      }
      className="emotion-5"
      dataCy="spacingBox"
      tag="div"
    >
      <div
        className="emotion-4"
        data-cy="spacingBox"
      >
        <Flex
          align="flex-start"
          direction="row"
          gutterSize="none"
          justify="flex-start"
          wrap="nowrap"
        >
          <div
            className="emotion-3"
            data-cy="flex"
          >
            <FlexItem
              flex="grow"
            >
              <div
                className="emotion-2"
                data-cy="flexItem"
              >
                <MonospaceText
                  align="inherit"
                  color="var(--themeTextColorPrimaryInverted, #FFFFFF)"
                  size="m"
                  tag="code"
                  weight="medium"
                  wrap="wrap"
                >
                  <Text
                    align="inherit"
                    className="emotion-1"
                    color="var(--themeTextColorPrimaryInverted, #FFFFFF)"
                    dataCy="monospaceText"
                    size="m"
                    tag="code"
                    weight="medium"
                    wrap="wrap"
                  >
                    <code
                      className="rmTextMargins emotion-0"
                      data-cy="monospaceText"
                    >
                      snippet content
                    </code>
                  </Text>
                </MonospaceText>
              </div>
            </FlexItem>
          </div>
        </Flex>
      </div>
    </Box>
  </SpacingBox>
</CodeSnippet>
`;

exports[`CodeSnippet renders with copy button 1`] = `
.emotion-5 {
  cursor: pointer;
}

.emotion-11 {
  overflow: auto;
  white-space: pre;
}

.emotion-10 {
  overflow: auto;
  white-space: pre;
  padding: 16px;
}

.emotion-9 {
  background-color: #1B2029;
  background-repeat: no-repeat;
  overflow: auto;
  white-space: pre;
  padding: 16px;
}

.emotion-8 {
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  height: auto;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 0;
}

.emotion-8 > div {
  width: auto;
}

.emotion-8 > *:not(:first-child) {
  padding-left: 0;
  padding-top: 0;
}

.emotion-2 {
  box-sizing: border-box;
  -webkit-flex-basis: 0;
  -ms-flex-preferred-size: 0;
  flex-basis: 0;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
  min-width: 0;
  width: auto;
}

.emotion-1 {
  font-family: 'Menlo','Bitstream Vera Sans Mono','DejaVu Sans Mono','Monaco','Consolas',monospace;
}

.emotion-0 {
  font-family: 'Menlo','Bitstream Vera Sans Mono','DejaVu Sans Mono','Monaco','Consolas',monospace;
  line-height: 1.4;
  font-weight: 500;
  font-size: 14px;
  color: var(--themeTextColorPrimaryInverted,#FFFFFF);
  fill: var(--themeTextColorPrimaryInverted,#FFFFFF);
  text-align: inherit;
}

.emotion-7 {
  box-sizing: border-box;
  -webkit-flex-basis: auto;
  -ms-flex-preferred-size: auto;
  flex-basis: auto;
  -webkit-box-flex: 0;
  -webkit-flex-grow: 0;
  -ms-flex-positive: 0;
  flex-grow: 0;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  width: initial;
}

.emotion-6 {
  background-repeat: no-repeat;
  display: inline-block;
}

.emotion-4 {
  background-repeat: no-repeat;
  display: inline-block;
  cursor: pointer;
}

.emotion-3 {
  vertical-align: middle;
  fill: var(--themeTextColorPrimaryInverted,#FFFFFF);
}

.emotion-3 use {
  pointer-events: none;
}

<CodeSnippet
  textToCopy="snippet content"
>
  <SpacingBox
    bgColor="#1B2029"
    className="emotion-11"
    side="all"
    spacingSize="m"
    tag="div"
  >
    <Box
      bgColor="#1B2029"
      bgImageOptions={
        Object {
          "position": undefined,
          "repeat": undefined,
          "size": undefined,
        }
      }
      className="emotion-10"
      dataCy="spacingBox"
      tag="div"
    >
      <div
        className="emotion-9"
        data-cy="spacingBox"
      >
        <Flex
          align="flex-start"
          direction="row"
          gutterSize="none"
          justify="flex-start"
          wrap="nowrap"
        >
          <div
            className="emotion-8"
            data-cy="flex"
          >
            <FlexItem
              flex="grow"
            >
              <div
                className="emotion-2"
                data-cy="flexItem"
              >
                <MonospaceText
                  align="inherit"
                  color="var(--themeTextColorPrimaryInverted, #FFFFFF)"
                  size="m"
                  tag="code"
                  weight="medium"
                  wrap="wrap"
                >
                  <Text
                    align="inherit"
                    className="emotion-1"
                    color="var(--themeTextColorPrimaryInverted, #FFFFFF)"
                    dataCy="monospaceText"
                    size="m"
                    tag="code"
                    weight="medium"
                    wrap="wrap"
                  >
                    <code
                      className="rmTextMargins emotion-0"
                      data-cy="monospaceText"
                    >
                      snippet content
                    </code>
                  </Text>
                </MonospaceText>
              </div>
            </FlexItem>
            <FlexItem
              flex="shrink"
            >
              <div
                className="emotion-7"
                data-cy="flexItem"
              >
                <ClickToCopyButton
                  color="var(--themeTextColorPrimaryInverted, #FFFFFF)"
                  textToCopy="snippet content"
                  tooltipContent="Copied to clipboard"
                >
                  <ClickToCopy
                    textToCopy="snippet content"
                    tooltipContent="Copied to clipboard"
                  >
                    <Box
                      bgImageOptions={
                        Object {
                          "position": undefined,
                          "repeat": undefined,
                          "size": undefined,
                        }
                      }
                      dataCy="box"
                      display="inline-block"
                      tag="div"
                    >
                      <div
                        className="emotion-6"
                        data-cy="box"
                      >
                        <Tooltip
                          id="copyTooltip1"
                          open={false}
                          suppress={true}
                          trigger={
                            <Clickable
                              action={[Function]}
                              disableFocusOutline={false}
                              role="button"
                              tabIndex={0}
                            >
                              <Box
                                bgImageOptions={
                                  Object {
                                    "position": undefined,
                                    "repeat": undefined,
                                    "size": undefined,
                                  }
                                }
                                dataCy="box"
                                display="inline-block"
                                tag="span"
                              >
                                <Icon
                                  color="var(--themeTextColorPrimaryInverted, #FFFFFF)"
                                  shape="system-clipboard"
                                />
                              </Box>
                            </Clickable>
                          }
                        >
                          <span
                            data-cy="tooltip"
                            onBlur={[Function]}
                            onFocus={[Function]}
                            onMouseLeave={[Function]}
                            onMouseOver={[Function]}
                          >
                            <Dropdownable
                              dropdown={
                                <TooltipContent
                                  id="copyTooltip1"
                                  open={false}
                                >
                                  Copied to clipboard
                                </TooltipContent>
                              }
                              open={false}
                              preferredDirections={
                                Array [
                                  "top",
                                  "top-start",
                                  "top-end",
                                  "bottom",
                                  "bottom-start",
                                  "bottom-end",
                                ]
                              }
                            >
                              <div>
                                <Clickable
                                  action={[Function]}
                                  disableFocusOutline={false}
                                  role="button"
                                  tabIndex={0}
                                >
                                  <Box
                                    bgImageOptions={
                                      Object {
                                        "position": undefined,
                                        "repeat": undefined,
                                        "size": undefined,
                                      }
                                    }
                                    className="emotion-5"
                                    dataCy="box"
                                    display="inline-block"
                                    onClick={[Function]}
                                    onKeyPress={[Function]}
                                    role="button"
                                    tabIndex={0}
                                    tag="span"
                                  >
                                    <span
                                      className="emotion-4"
                                      onClick={[Function]}
                                      onKeyPress={[Function]}
                                      role="button"
                                      tabIndex={0}
                                    >
                                      <Icon
                                        color="var(--themeTextColorPrimaryInverted, #FFFFFF)"
                                        shape="system-clipboard"
                                      >
                                        <svg
                                          aria-label="system-clipboard icon"
                                          className="emotion-3"
                                          data-cy="icon"
                                          height={24}
                                          preserveAspectRatio="xMinYMin meet"
                                          role="img"
                                          viewBox="0 0 24 24"
                                          width={24}
                                        >
                                          <use
                                            xlinkHref="#system-clipboard"
                                          />
                                        </svg>
                                      </Icon>
                                    </span>
                                  </Box>
                                </Clickable>
                              </div>
                            </Dropdownable>
                          </span>
                        </Tooltip>
                      </div>
                    </Box>
                  </ClickToCopy>
                </ClickToCopyButton>
              </div>
            </FlexItem>
          </div>
        </Flex>
      </div>
    </Box>
  </SpacingBox>
</CodeSnippet>
`;
